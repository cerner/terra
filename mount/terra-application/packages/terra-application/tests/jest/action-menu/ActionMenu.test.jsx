import React from 'react';

import ActionMenu from '../../../src/action-menu/ActionMenu';

const MockComponent = () => <li>item</li>;
MockComponent.interactiveType = false;

describe('ActionMenu', () => {
  it('should render with minimal props', () => {
    const wrapper = shallow((
      <ActionMenu label="test menu" />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with all props', () => {
    const wrapper = shallow((
      <ActionMenu
        label="test menu"
        isHeaderDisplayed
        onRequestClose={jest.fn()}
        isHeightBounded
        isWidthBounded
      >
        <MockComponent />
      </ActionMenu>
    ));
    expect(wrapper).toMatchSnapshot();
  });
});
