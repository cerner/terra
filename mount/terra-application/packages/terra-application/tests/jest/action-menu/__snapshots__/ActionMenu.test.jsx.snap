// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionMenu should render with all props 1`] = `
<ContentContainer
  aria-label="test menu"
  className="action-container"
  fill={true}
  header={
    <injectIntl(ActionMenuHeader)
      label="test menu"
      onClose={[MockFunction]}
    />
  }
  onKeyDown={[Function]}
  role="dialog"
  tabIndex="0"
>
  <ul
    aria-label="test menu"
    className="action-menu"
    onKeyDown={[Function]}
    role="menu"
    tabIndex="0"
  >
    <MockComponent
      indentChildren={false}
      key=".0"
      onArrow={[Function]}
      onChar={[Function]}
    />
  </ul>
</ContentContainer>
`;

exports[`ActionMenu should render with minimal props 1`] = `
<ul
  aria-label="test menu"
  className="action-menu"
  onKeyDown={[Function]}
  role="menu"
  tabIndex="0"
/>
`;
